core/src/ast_builder/select/root.rs

```diff
@@ -61,29 +62,76 @@ impl SelectNode {
         )
     }
 
-    pub fn left_join<'a>(self, table_name: &str) -> JoinNode<'a> {
+    pub fn left_join(self, table_name: &str) -> JoinNode<'a> {
         JoinNode::new(self, table_name.to_owned(), None, JoinOperatorType::Left)
     }
 
-    pub fn left_join_as<'a>(self, table_name: &str, alias: &str) -> JoinNode<'a> {
+    pub fn left_join_as(self, table_name: &str, alias: &str) -> JoinNode<'a> {
         JoinNode::new(
             self,
             table_name.to_owned(),
             Some(alias.to_owned()),
             JoinOperatorType::Left,
         )
     }
+
+    pub fn alias_as(self, table_alias: &'a str) -> TableAliasNode {
+        let table_node = TableNode {
+            table_name: table_alias.to_owned(),
+            table_type: TableType::Derived {
+                subquery: Box::new(self.clone()),
+                alias: table_alias.to_owned(),
+            },
+            args: self.table_node.args,
+        };
+
+        TableAliasNode {
+            table_node,
+            table_alias: table_alias.to_owned(),
+        }
+    }
 }
 
-impl Prebuild for SelectNode {
+impl<'a> Prebuild for SelectNode<'a> {
     fn prebuild(self) -> Result<NodeData> {
-        let relation = TableFactor::Table {
-            name: self.table_name,
-            alias: self.table_alias.map(|name| TableAlias {
-                name,
+        let alias = self.table_alias.map(|name| TableAlias {
+            name,
+            columns: Vec::new(),
+        });
+
+        let alias_or_name = match alias.clone() {
+            Some(alias) => alias,
+            None => TableAlias {
+                name: self.table_node.table_name.clone(),
                 columns: Vec::new(),
-            }),
-            index: None,
+            },
+        };
+
+        let relation = match (self.table_node.table_type, self.table_node.args) {
+            (TableType::Table, _) => TableFactor::Table {
+                name: self.table_node.table_name,
+                alias,
+                index: None,
+            },
+            (TableType::Dictionary(dict), _) => TableFactor::Dictionary {
+                dict,
+                alias: alias_or_name,
+            },
+            (TableType::Series, Some(args)) => TableFactor::Series {
+                alias: alias_or_name,
+                size: args.try_into()?,
+            },
+            (TableType::Series, None) => unreachable!(),
+            (TableType::Derived { subquery, alias }, _) => match subquery.build()? {
+                Statement::Query(subquery) => TableFactor::Derived {
+                    subquery,
+                    alias: TableAlias {
+                        name: alias,
+                        columns: Vec::new(),
+                    },
+                },
+                _ => unreachable!(),
```

Comment:
I got 2 unreachable cases which look like would filtered by parseError in advance.